cookbooks/aws-parallelcluster-slurm/files/default/config_slurm/scripts/health_check_manager.py

```diff
@@ -0,0 +1,375 @@
+                        )
+                        health_checks.append(health_check)
+
+        except (KeyError, TypeError) as err:
+            if hasattr(err, "message"):
+                err = err.message
+            log.warning("Ignore missing %s in config", err)
+
+        health_check_conf = HealthCheckConfig(
+            region=args.region,
+            stack_name=args.stack_name,
+            node_type=args.node_type,
+            queue_name=args.queue_name,
+            compute_resource_name=args.compute_resource_name,
+            health_checks=health_checks,
+        )
+
+        log.debug("HealthCheck config %s: ", health_check_conf)
+        return health_check_conf
+
+
+def _execute_health_checks(health_check_manager_config: HealthCheckManagerConfig, args: List[str]) -> int:
+    """Execute all Health Check."""
+    health_check_conf = HealthCheckConfigLoader().load_configuration(health_check_manager_config, args)
+
+    exit_code_sum = 0
+
+    for health_check in health_check_conf.health_checks:
+        if health_check.is_enabled:
+            try:
+                log.info(
+                    "Executing Health Check '%s' for queue '%s' and compute resource '%s'",
+                    health_check.name,
+                    health_check_conf.queue_name,
+                    health_check_conf.compute_resource_name,
+                )
+
+                # The command in this subprocess call is built as literal
+                result = subprocess.run(
+                    health_check.check_path,
+                    timeout=health_check_manager_config.health_check_timeout,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.STDOUT,
+                    encoding="utf-8",
+                    check=False,
+                    shell=False,  # nosec B603
+                )
+                exit_code_sum += result.returncode
+                if result.stdout:
+                    log.info("Output of Health Check '%s' execution is:\n%s", health_check.name, result.stdout)
+                else:
+                    log.info("Output of Health Check '%s' execution is empty", health_check.name)
+            except (subprocess.SubprocessError, OSError) as err:
+                if hasattr(err, "message"):
+                    err = err.message
+                log.error(
+                    "Failure when executing Health Check '%s' for queue '%s' and compute resource '%s', with error: %s",
+                    health_check.name,
+                    health_check_conf.queue_name,
+                    health_check_conf.compute_resource_name,
+                    err,
+                )
+    if not health_check_conf.health_checks:
+        log.info("No Health Check enabled found")
+
+    return exit_code_sum
+
+
+def _parse_arguments() -> List[str]:
+    parser = argparse.ArgumentParser(description="Health Check Manager")
+    parser.add_argument(
+        "-r",
+        "--region",
+        type=str,
```

Comment:
minor: you can skip the type for `str` parameter because it is the default

Reply:
removed